frontend/src/components/shared/FeedbackDialog.tsx

```diff
@@ -0,0 +1,142 @@
+import React, { FormEvent, useState } from 'react';
+import Dialog, { DialogProps } from '@material-ui/core/Dialog';
+import DialogTitle from '@material-ui/core/DialogTitle';
+import { withStyles } from '@material-ui/core/styles';
+import Radio from '@material-ui/core/Radio';
+import RadioGroup from '@material-ui/core/RadioGroup';
+import FormControlLabel from '@material-ui/core/FormControlLabel';
+import Rating from '@material-ui/lab/Rating';
+import { Locale } from 'scenes/types';
+import { Connection } from 'twilio-client';
+import { PrimaryButton } from './RoundedButton';
+
+interface FeedbackDialogProps extends DialogProps {
+  connection: Connection;
+  locale: Locale;
+}
+
+// Reference for Problem Keys: https://www.twilio.com/docs/voice/sdks/javascript/overview-1x-deprecated/connection
+const EN_STRINGS = {
+  FEEDBACK_DIALOG_TITLE: 'How was your call?',
+  FEEDBACK_SUBMIT_LABEL: 'Submit',
+  FEEDBACK_TITLES: {
+    GOOD: 'GOOD',
+    BAD: 'BAD',
+  },
+  PROBLEM_TITLES: {
+    'Call got disconnected': Connection.FeedbackIssue.DroppedCall,
+    'Lag in audio': Connection.FeedbackIssue.AudioLatency,
+    "Can't hear one another": Connection.FeedbackIssue.OneWayAudio,
+    'Sound breaking up': Connection.FeedbackIssue.ChoppyAudio,
+    'Poor audio quality': Connection.FeedbackIssue.NoisyCall,
+    'Audible echos during call': Connection.FeedbackIssue.Echo,
+  },
+};
+
+const STRINGS = {
+  en: EN_STRINGS,
+  bn: {
+    ...EN_STRINGS,
+  },
+};
+
+// TODO move this to its own module
+const withDialogButtonStyles = withStyles(() => ({
+  root: {
+    padding: '1em 2em',
+    flex: '1 0',
+    margin: '0 0.5rem',
+  },
+}));
+
+const DialogPrimaryButton = withDialogButtonStyles(PrimaryButton);
+
+export default function FeedbackDialog({
+  connection,
+  locale,
+  open,
+  onClose,
+}: FeedbackDialogProps) {
+  const [feedback, setFeedback] = useState(-1);
+  const [hover, setHover] = useState(-1);
+  const [problem, setProblem] = useState<Connection.FeedbackIssue | undefined>(
+    Connection.FeedbackIssue.DroppedCall
```

Comment:
Why is the default DroppedCall instead of undefined?

Reply:
React complained that we are making an uncontrolled variable controlled when setting it to undefined at the start. Unless you are okay with it?

Reply:
Hm, I don't see why this would be a React issue. In either case, we are controlling the form. Could you try setting it to `undefined` and show me the warning/error output that you're seeing?

Reply:
![image](https://user-images.githubusercontent.com/29354437/132233489-598d6026-8651-48ee-86eb-2e4a138b079d.png)

Reply:
It still works fine tho, despite the warning

Reply:
It looks like you found a good solution here, which is to use `null` instead of `undefined`. Then there isn't even a need to cast, which is great :)